### PR TITLE
Delay showing the dialog until the next editor gui update

### PR DESCRIPTION
**Pull request by StephenHodgson**
# XRTK - Mixed Reality Toolkit Pull Request

## Overview

Delay showing the dialog for no profile when setting up to prevent dialog cancellation on mac

### CODE REVIEW

File: XRTK-Core/Packages/com.xrtk.core/Editor/MixedRealityToolkitInspector.cs [hunk 1]
```diff
@@ -76,7 +76,8 @@ public override void OnInspectorGUI()
                 {
                     case 0:
                         EditorGUIUtility.PingObject(target);
-                        EditorUtility.DisplayDialog("Attention!", "No root profile for the Mixed Reality Toolkit was found.\n\nYou'll need to create a new one.", "OK");
+                        EditorApplication.delayCall += () =>
```
Comment (FejZa) [inline XRTK-Core/Packages/com.xrtk.core/Editor/MixedRealityToolkitInspector.cs:79]: I am not sure with these .delayCall registration. Same thing as with the package installer we talked about yesterday. Pretty sure this is causing memory leaks in the editor.

Comment (StephenHodgson) [reply XRTK-Core/Packages/com.xrtk.core/Editor/MixedRealityToolkitInspector.cs:79]: They're just closures. Should be okay.

It's a lot harder to create memory leaks I'm c# unless you're playing with low level memory

Comment (StephenHodgson) [reply XRTK-Core/Packages/com.xrtk.core/Editor/MixedRealityToolkitInspector.cs:79]: @FejZa can you find me an example usage of this delay call that doesn't use an anonymous delegate?

Comment (FejZa) [reply XRTK-Core/Packages/com.xrtk.core/Editor/MixedRealityToolkitInspector.cs:79]: Like in XRTK? E.g. here https://github.com/XRTK/WindowsMixedReality/blob/development/XRTK.WindowsMixedReality/Packages/com.xrtk.wmr/Editor/WindowsMixedRealityPackageInstaller.cs

Comment (FejZa) [reply XRTK-Core/Packages/com.xrtk.core/Editor/MixedRealityToolkitInspector.cs:79]: I don't know. To me this looks like every GUI frame we are registering an event handler. But I might be wrong here. I just live by the rule, if I subscribe, I unsubscribe.

**Review by FejZa — APPROVED** [review]

File: XRTK-Core/Packages/com.xrtk.core/Editor/MixedRealityToolkitInspector.cs (hunk 1 above)
Comment (StephenHodgson) [reply XRTK-Core/Packages/com.xrtk.core/Editor/MixedRealityToolkitInspector.cs:79]: No just in general.

Comment (StephenHodgson) [reply XRTK-Core/Packages/com.xrtk.core/Editor/MixedRealityToolkitInspector.cs:79]: Let's take this discussion to discord :)

**Comment by FejZa** [comment]
Well, should have read the docs:

"Delegate which is called once after all inspectors update.
Add functions to this delegate in order to delay their execution until after inspectors have updated. Each function is only executed once after it is added."

It seems Unity removes every registered method once called automatically.

**Comment by StephenHodgson** [comment]
> It seems Unity removes every registered method once called automatically.

That was my understanding as well

**Comment by StephenHodgson** [comment]
image

**Pull request merged**